### ISSUE TITLE
Define invalid message object

### ISSUE THREAD

**Issue by samends** [dojo/widgets#1614]
**Enhancement**

Currently the interface for an invalid message object is neither defined nor exported. The invalid message object is part of the definition of `Validity` found at https://github.com/dojo/widgets/blob/master/src/form/middleware.ts#L4 however more time than not projects have to write their own interfaces for `{ valid: boolean, message: string }`. Perhaps we could define and export a new interface for this with the name `InvalidMessage`

Package Version: 7.0.3

**Comment by tomdye** [comment]
I don't think we should be exporting this interface. Perhaps we should instead change to always export the same object with a valid boolean and optional message. This needs investigating further.

**Comment by samends** [comment]
@tomdye you mean change the invalid type on the widgets to only be `{ valid: boolean, message: string }` and not `{ valid: boolean, message: string } | true` that would certainly lessen alot of logic that has to go in to handle two different types

**Comment by tomdye** [comment]
Correct, I'm not sure why it was done that way in the first place. I'm just thinking out loud here though - would this approach work for all our validated widgets?

**Comment by samends** [comment]
I think so, I'm also not sure why it's currently `{ valid: boolean, message: string } | true`, it seems to really complicate the logic for not alot of gain

**Comment by samends** [comment]
I took at quick look at where this is used in the code base and it seems like 4 places would need to get updated:
- `TextInput`'s valid
- `TextArea`'s valid
- `NumberInput`'s valid
- Form's middleware valid() https://github.com/dojo/widgets/blob/master/src/form/middleware.ts#L106
